Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm

```diff
@@ -238,4 +243,32 @@
     return [description stringByAppendingString:layerDescription];
 }
 
+#if ENABLE(THREADED_ANIMATION_RESOLUTION)
+void RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues(const WebCore::AcceleratedEffects& effects, const WebCore::AcceleratedEffectValues& baseValues, RemoteLayerTreeHost* host)
+{
+    ASSERT(isUIThread());
+    ASSERT(host);
+
+    if (m_effects)
+        host->animationEffectStackWasRemoved(*m_effects);
+
+    if (effects.isEmpty())
+        return;
+
+    m_effects = new RemoteAcceleratedEffectStack(layer().bounds, host->acceleratedTimelineTimeOrigin());
+
+    WebCore::AcceleratedEffects clonedEffects;
+    clonedEffects.reserveCapacity(effects.size());
+    for (auto& effect : effects)
+        clonedEffects.append(effect->clone());
```

Comment:
Do we need the deep-clone here? `AcceleratedEffects` is a `Vector` of `Refs`, can we just copy the vector so that we get new strong refs to the same objects?